opensearch-operator/pkg/reconcilers/rollingRestart.go

```diff
@@ -190,12 +191,14 @@ func (r *RollingRestartReconciler) restartStatefulSetPod(sts *appsv1.StatefulSet
 		}, nil
 	}
 
-	err = r.Delete(r.ctx, &corev1.Pod{
+	eviction := &policyv1.Eviction{}
```

Comment:
suggesting to keep upstream code commented (with /* */ block to avoid as much as possible conflicts when bumping operator), and to re-explain here rational of our change, ie using eviction instead of Delete to respect PDB during roll-restart and with advantage of fixing concurrent roll-restart on node pools. A reference to upstream issue would be nice too.

Reply:
Done

Reply:
Undone !